### ISSUE TITLE
bin/magento dev : email:override-compatibility-check or bin/magento dev : email:newsletter-compatibility-check give false positives

### ISSUE THREAD

**Issue by leonhelmus** [magento/magento2#35343]
### Preconditions (*)

1. Magento 2.4.3-p2 on premise enterprise/open source
2. php 7.4

### Steps to reproduce (*)

1. Update from magento 2.4.3-p1 to Magento 2.4.3-p2
2. Have a mail template in the backend with either {{depend customer.getFirstName()}} or {{if customer.getFirstName()}} or {{depend order.getExtensionAttributes().getExpectedDeliveryDateFrom()}} or {{trans ""Estimated Delivery Date:""}} {{trans ""Between %date_from and %date_to"" date_from=$order.getExtensionAttributes().getCustomExample() date_to=$order.getExtensionAttributes().getCustomExample()}} for order email.
3. This is not valid syntax in the non-legacy (using reference: https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/templates/template-email-migration.html) flows but the bin/magento dev : email:override-compatibility-check or bin/magento dev : email:newsletter-compatibility-check check doesn't pick up on it (for me at least). When the email is sent by the system you'll be stuck with empty values here. 

### Expected result (*)

1. bin/magento dev:email :override-compatibility-check or bin/magento dev: email:newsletter-compatibility-check should give a warning.

### Actual result (*)

1.  bin/magento dev:email :override-compatibility-check or bin/magento dev: email:newsletter-compatibility-check should no warning is shown.

### Workaround
1. Check all email templates by hand, since the tool cannot completely be trusted and takes a lot of time. You have to look in app/design, backend email templates/ newsletter templates, vendor modules. Which is quite a lot of work and patch takes longer than expected!

I have also heard that if you do the same in app/design it does not check as well from colleagues does not check there is the theme is not extended from Luma. Also im not sure if it checks vendor email templates. Used as a reference: https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/templates/template-email-migration.html 

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [x] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by nathanjosiah** [comment]
@magento export issue to Jira project AC as Bug

**Comment by nathanjosiah** [comment]
Actually, reading through this again this isn't a bug. This is a static analyzer that doesn't know the data that could be used with the template. Take a look at the usage options in https://devdocs.magento.com/guides/v2.4/frontend-dev-guide/templates/template-email-migration.html#data-objects-and-geturl Specifically that `DataObjects` are still allowed to use the `object.getXYZ()` format for backwards compatibility. Because of this, a static tool has no way to know if `object` (in this example) is an instance of `DataObject` or something else.

**Comment by leonhelmus** [comment]
So your saying it does not know what template vars can be used? We had the same issue with the layout hanle in mails that send a order object instead of an order id. It also gave no issue.

I thought it would show all wrong examples and would tell you what is wrong. 

What is the command than used for, because I missed a lot of data. For example with order.getIsNotVirtual versus order_data.is_not_virtual.

**Comment by nathanjosiah** [comment]
It's not about the template vars that can be used. If the template vars define invalid usage it will also be flagged. Let's look at an example, let's say you have an object
```
class Foo {
   public function getThing() { return 'abc'; }
}
```
Using this object in a template is invalid (`{{var foo.getThing()}}`). It's a regular object, method access is not allowed.

But, let's say the object looks like this instead:

```
class Foo extends DataObject {}
...
$foo->setData('thing', 'abc');
```

Using this object in a template can have identical usage (`{{var foo.getThing()}}`). It's DataObject so pseudo-method access is allowed. The variable resolver will ultimately call `foo.getData('thing')` but the `foo.getThing()` format is allowed as syntactic sugar for backwards compatibility.

However, the static resolver only has the template to look at when it is evaluating the templates. Email templates can be used for anything and have any data given to them. Because of these things, there is no way to know if that usage is invalid or not. Since we fixed all of the default core templates to be compatible with the strict resolver we determined it would be uncommon for a getter method to be used on a non-DataObject and opted to not flood the output of the analyzer with unhelpful feedback about _possible_ getter usage. Maybe we can enable this in the future behind a flag or something.
